src/Swashbuckle.AspNetCore.Cli/Program.cs

```diff
@@ -165,14 +172,22 @@ public static int Main(string[] args)
                         }
                     }
 
-
                     using Stream stream = outputViaConsole ? Console.OpenStandardOutput() : File.Create(outputPath);
-                    using StreamWriter writer = new StreamWriter(stream, outputViaConsole ? Console.OutputEncoding : Encoding.UTF8);
+                    using StreamWriter writer = new(stream, outputViaConsole ? Console.OutputEncoding : Encoding.UTF8);
 
-                    writer.WriteLine($"Swagger Document Names (Surrounded by single quotes):");
-                    foreach (var name in docNames)
+                    if (docMetaProvider == null)
                     {
-                        writer.WriteLine($"'{name}'");
+                        writer.WriteLine($"ERROR: ISwaggerProvider instance DOES NOT support ISwaggerDocumentMetadaatProvider, so this coommand "
+                            + "DOES NOT support listing out the document names. Try updating the version of Swashbuckle used by the project you are examining to a newer version.");
```

Comment:
We already checked whether it wasn't null earlier, so the error can be emitted there and stop before doing any more work.

```csharp
if (swaggerProvider is not ISwaggerDocumentMetadataProvider docMetaProvider)
{
    // error message
    return -1;
}

var docNames = docMetaProvider.GetDocumentNames();

// etc.
```